backend/app/services/dingtalk/docs_service.py

```
@@ -0,0 +1,450 @@
+
+        except httpx.HTTPStatusError as e:
+            logger.error(
+                f"HTTP error from DingTalk MCP: {e.response.status_code} - {e.response.text}"
+            )
+            if e.response.status_code == 401 or e.response.status_code == 403:
+                raise ValueError(
+                    f"DingTalk authentication required. Please ensure:\n"
+                    f"1. You have access to this document in DingTalk\n"
+                    f"2. The document is shared with you or is public\n"
+                    f"3. Your DingTalk MCP configuration is correct"
+                )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)